### ISSUE TITLE
economy system

### ISSUE THREAD

**Issue by yoel123** [yoel123/java-text-based-space-station-manger-sim#10]
for now, its random, item prices and amounts go up and down without reason, while it is roughly based on supply and demand, its still just random.

I don't want to simulate something realistic and not fun, I want a challenging system.

maybe a trade relations system, like if you have a trading agreement with the cheapest fuel manufacturer in the sector you will get cheap fuel prices more often.

or the prices will drastically go up if there's a pirate raid or another event.

if the station is close to an ore-rich system with mining operations naturally ore will be cheaper most of the time.

there shod also be a market prediction system (like which item will go up or down in price next turn) so the player can pay to predict this and plan which goods to buy cheap and sell high.

maybe upgrades that affect the amount of goods coming in like market size, like too meany ships wont come to trade if thers no room for their goods (their not gonna wait outside for their turn usually).

its as easy as checking if(upgrades.contain("big market")) then goods amount+=2

stability of the station shod also affect goods arriving, no one wants to bring his trade to a den of thieves just to have his cargo stolen at gunpoint after he docked (thats for bigger stations and depended on the number of guards and the constable character and other stuff).

maybe let the player have a small trade fleet to send on missions. this shod also affect prices.

**Comment by rc14193** [comment]
I’m personally interested in working on this. So let me see if I have so far what the price equation should be based on what you’ve listed so far: supply+demand+station stability+any system buffs+trade agreement modifier+Event modifier. Here supply can be based on goods in market, demand I’m guessing will have to stay random. The rest will vary based on game conditions so that’ll be the layout. Anything I’m missing?

Also if I’m understanding you then you also want the equivalent of put and call options like a stock market for goods?

**Comment by yoel123** [comment]
you got it right, all of this is in supply and demand method in the market class.

about the "actual" ingame  supply+demand, it kind of fluctuates randomly because you have ships from all over passing through, so one can come from an ore-rich world and desperately want to buy food, while the other comes from a capital world and want luxury goods, while yet another ship comes from the main fuel manufacture, brings a lot of fuel but only wants to buy weapons.

a trader doesn't want to fly with an empty ship, so as soon as he sales his cargo, he wants to buy more for another destination of his trade route, as a trader, you don't want money that "doesn't work" or money in the shape of unsold goods so you sell them ,loss money, and with that money, you buy other goods that make profit that cover the loss and more.

in short, you cant control the supply you get or the demand, unless it's in your warehouse/cargo bay and not for sale in the market.

oh and you don't make money of market tariffs, that goes to the faction that helped you build and protect the station.

you can kind of think of it as a stock market, your holding a percentage of all the sectors goods. and they go up and down.

prices shouldn't change dramatically except for special events (like system pirate raid, famine, etc).

**Comment by rc14193** [comment]
Adding some of our notes and asking another question. Resources are as follows: Food, Supply, Fuel, Credits,Ships, Morality.

In the first post you mentioned ore. Should that be added? Is that going to be the main trade able good? The player will mine for ore and sell that to merchants for credits? I see all the stuff to buy but not clear now on how the player earns money. As it stands now I only see resources the player needs to buy, nothing they can sell to keep themselves profitable.

**Comment by yoel123** [comment]
I don't plan for the close future for the player to manufacture any goods.
maybe a hydroponic farm upgrades to use when the station runs out of food but not for manufacturing food.

to clarify most goods are used only for trading you don't need them for anything.

if you start manufacturing any type of goods you will hold a percentage of the market, something that might piss-off some factions that like to have a monopoly on certain goods. and will result in an attack to cripple your production for a while (it's a cruel sector).

trade goes like this the player buys goods this turn, the next turn the goods price might go up, if so sell the goods and make a profit (after taxes and maintenance, salaries, and insurance costs of course). 

for example: you buy 200 lobsters for 20 credits, next turn the lobster price goes up to 30 credits, you sell the lobsters for 50% profit (after all tax and expenses).

the other way to make credits is to build rental spaces (but trading can be more profitable especially if you buy weapons from a renegade general event for medium-low price and sell them in the market at a higher price for profit).
